src/main/java/duke/utils/Parser.java

```diff
@@ -0,0 +1,201 @@
+     * @param ui the object in charge of printing user-friendly outputs.
+     * @param storage the object in charge of writing to the local storage file.
+     */
+    public Parser(TaskList taskList, Ui ui, Storage storage) {
+        this.taskList = taskList;
+        this.ui = ui;
+        this.storage = storage;
+    }
+
+    /**
+     * Returns command associated with the command line input from user.
+     *
+     * @param input command line input from user.
+     * @return command associated with input from user.
+     * @throws EmptyArgumentException when only a 1 word command is passed without any following input.
+     * @throws InvalidDateTimeException when date entered by user is not a valid date or not an acceptable date format.
+     * @throws InvalidIndexInputException when index entered by user is not a number or not within range of 1 to
+     *     the size of the TaskList.
+     * @throws EmptyListException when trying to find by keyword but TaskList is empty.
+     */
+    public Command parse(String input) throws EmptyArgumentException, InvalidDateTimeException,
+            InvalidIndexInputException, EmptyListException {
+        String[] commandAndInput = input.split(" ", 2);
+        String command = commandAndInput[0];
+
+        switch (command) {
+        case ToDoCommand.COMMAND_WORD:
+            return prepareToDo(commandAndInput);
+
+        case DeadlineCommand.COMMAND_WORD:
+            return prepareDeadline(commandAndInput);
+
+        case EventCommand.COMMAND_WORD:
+            return prepareEvent(commandAndInput);
+
+        case FindCommand.COMMAND_WORD:
+            return prepareFind(commandAndInput);
+
+        case DoneCommand.COMMAND_WORD:
+            return prepareDone(commandAndInput);
+
+        case DeleteCommand.COMMAND_WORD:
+            return prepareDelete(commandAndInput);
+
+        case ListCommand.COMMAND_WORD:
+            return prepareList();
+
+        case ByeCommand.COMMAND_WORD:
+            return prepareExit();
+
+        default:
+            return prepareHelp();
+        }
+    }
+
+    private Command prepareToDo(String[] arguments) {
+        return new ToDoCommand(this.taskList, this.ui, this.storage, arguments[1]);
+    }
+
+    private Command prepareDeadline(String[] arguments) throws EmptyArgumentException, InvalidDateTimeException {
+        if (arguments.length == 1) {
+            throw new EmptyArgumentException("Please input a valid task description!");
+        } else {
+            String description = arguments[1];
+            String[] taskInputAndDate = description.split("/", 2);
+
+            taskInputAndDate[0] = taskInputAndDate[0].trim();
+            taskInputAndDate[1] = taskInputAndDate[1].trim();
+
+            try {
+                LocalDateTime dateTime = LocalDateTime.parse(taskInputAndDate[1].substring(3), FORMATTER);
+                return new DeadlineCommand(this.taskList, this.ui, this.storage, taskInputAndDate[0], dateTime);
```

Comment:
There are many parameters to this DeadlineCommand object. Perhaps by breaking after a comma would improve the readability of the code?